### PR TITLE
Refactored integration test for ContentService

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | N/A
| **Required by** | #2753
| **Improvement**| yes
| **New feature**    | no
| **Target version** | `7.5` 
| **BC breaks**      |no
| **Tests pass**     | yes
| **Doc needed**     | no

This refactor is a result of working on ContentService and a need to add some new tests and change the old ones in https://jira.ez.no/browse/EZP-30427

**TODO**:
- [x] Refactor tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 1]
```diff
@@ -743,7 +695,7 @@ public function testLoadVersionInfoById()
      * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testLoadVersionInfoById
      * @covers \eZ\Publish\Core\Repository\ContentService::loadVersionInfoById
      *
-     * @param \eZ\Publish\API\Repository\Values\Content\VersionInfo $versionInfo
+     * @param VersionInfo $versionInfo
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: Still using FQCN here

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 2]
```diff
@@ -6655,24 +6165,23 @@ public function testCopyTranslationsFromPublishedToDraft()
...
      *
-     * @param \eZ\Publish\API\Repository\LocationService $locationService
      * @param int $parentLocationId
      *
      * @return \eZ\Publish\API\Repository\Values\Content\Location[] A list of Locations aimed to be parents
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
-     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws NotFoundException
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: FQCN drop detected

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 3]
```diff
@@ -6317,56 +5849,53 @@ function (Location $parentLocation) use ($locationService) {
      * @covers \eZ\Publish\API\Repository\ContentService::revealContent
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
-     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws NotFoundException
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: FQCN drop detected

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 4]
```diff
@@ -6143,12 +5682,11 @@ private function createLocaleFieldsFixture($languageCode)
     /**
      * Asserts that given Content has default ContentStates.
      *
-     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     * @param ContentInfo $contentInfo
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: FQCN drop detected

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 5]
```diff
@@ -6270,45 +5805,42 @@ private function getExpectedMediaContentInfoProperties()
      * @covers \eZ\Publish\API\Repository\ContentService::hideContent
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
-     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws NotFoundException
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: FQCN drop detected

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 6]
```diff
@@ -542,7 +518,7 @@ public function testLoadContentInfo()
      * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testLoadContentInfo
      * @covers \eZ\Publish\API\Repository\ContentService::loadContentInfo
      *
-     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     * @param ContentInfo $contentInfo
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: FQCN drop detected

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 7]
```diff
@@ -637,7 +601,7 @@ public function testLoadContentInfoByRemoteId()
      * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testLoadContentInfoByRemoteId
      * @covers \eZ\Publish\API\Repository\ContentService::loadContentInfoByRemoteId
      *
-     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     * @param ContentInfo $contentInfo
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: FQCN drop detected

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 8]
```diff
@@ -3242,7 +2967,7 @@ public function testLoadVersions()
      * @depends eZ\Publish\API\Repository\Tests\ContentServiceTest::testLoadVersions
      * @covers \eZ\Publish\Core\Repository\ContentService::loadVersions
      *
-     * @param \eZ\Publish\API\Repository\Values\Content\VersionInfo[] $versions
+     * @param VersionInfo[] $versions
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: FQCN drop detected

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 9]
```diff
@@ -5629,15 +5197,12 @@ public function providerForDeleteTranslationFromDraftRemovesUrlAliasOnPublishing
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
      * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
-     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
-     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws NotFoundException
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: FQCN

**Comment by pspanja** [comment]
Just for info - tests were intentionally written in that style, because the idea was to extract the parts between `/* BEGIN: Use Case */` and `/* END: Use Case */` with a script and use it to automatically generate API examples for documentation. With this change the tests are not usable for that any longer, as the code does not read easily if embedded without context. So you could remove the markers as well.

BTW, the extraction script should still be somewhere in the repository.

**Comment by alongosz** [comment]
Thanks @pspanja for your input! I didn't know that :)

**Comment by mikadamczyk** [comment]
Thank you for your explanation @pspanja :)

**Comment by alongosz** [comment]
No QA required - only integration test changes. Merging, thank you @mikadamczyk!

**Pull request merged**